crates/rpc/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -65,3 +67,88 @@ pub struct Transaction {
     #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
     pub transaction_type: Option<U64>,
 }
+
+impl Transaction {
+    /// Create a new rpc transction result, using the given block hash, number, and tx index fields
+    /// to populate the corresponing fields in the rpc result.
+    pub(crate) fn from_recovered_with_block_context(
```

Comment:
this is a mined transaction.
I like the name but lets include that info in the docs.

Reply:
added docs making it clear that this is for a mined transaction